src/main/java/com/carleton/CapstoneSystem/Controllers/SchedulingController.java

```diff
@@ -0,0 +1,75 @@
+package com.carleton.CapstoneSystem.Controllers;
+
+import com.carleton.CapstoneSystem.DTO.ScheduleDTO;
+
+import com.carleton.CapstoneSystem.models.*;
+import com.carleton.CapstoneSystem.repositories.CoordinatorRepository;
+import com.carleton.CapstoneSystem.repositories.ProfessorRepository;
+import com.carleton.CapstoneSystem.repositories.ProjectRepository;
+import com.carleton.CapstoneSystem.repositories.StudentRepository;
+
+import com.carleton.CapstoneSystem.utils.EmailSendingError;
+import com.carleton.CapstoneSystem.utils.ProjectErrorMessages;
+import com.carleton.CapstoneSystem.utils.SchedulingErrorMessages;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+
+
+@Controller
+public class SchedulingController {
+    @Autowired
+    StudentRepository studentRepository;
+    @Autowired
+    ProfessorRepository professorRepository;
+    @Autowired
+    CoordinatorRepository coordinatorRepository;
+    @Autowired
+    ProjectRepository projectRepository;
+
+
+
+    public Response submitSchedule(Principal principal, ScheduleDTO scheduleDTO) {
+        if(scheduleDTO == null) {
+            return Response.status(Response.Status.BAD_REQUEST).entity(SchedulingErrorMessages.EMPTY_SCHEDULE_INFO).build();
+        }
+
```

Comment:
we are supposed to be validating the scheduleDto too. What do you think?

Reply:
yea i believe so